tasks/pg_utils/database.py

```diff
@@ -11,93 +11,135 @@
 from contextlib import contextmanager
 import datetime
 import uuid
+# noinspection PyPackageRequirements
+from typing import Optional, Any, Union, List, Dict
+
+# noinspection PyPackageRequirements
 import boto3
 from psycopg2 import DataError, ProgrammingError
 from psycopg2 import connect as psycopg2_connect
 from psycopg2 import sql
 from psycopg2.extras import RealDictCursor
+from psycopg2.extensions import connection, cursor
 
 LOGGER = logging.getLogger(__name__)
 
-#TODO develop tests for database.py later. in those mock psycopg2.cursor, etc
+
+# TODO develop tests for database.py later. in those mock psycopg2.cursor, etc
 
 class DbError(Exception):
     """
     Exception to be raised if there is a database error.
     """
 
+
 class ResourceExists(Exception):
     """
     Exception to be raised if there is an existing database resource.
     """
 
-def get_utc_now_iso():
+
+def get_utc_now_iso() -> str:
     """
-    Returns the current utc timestamp as a string in isoformat
-    ex. '2019-07-17T17:36:38.494918'
+    Takes the current utc timestamp and returns it an isoformat string.
+
+    Returns:
+        An isoformat string.
+        ex. '2019-07-17T17:36:38.494918'
     """
     return datetime.datetime.utcnow().isoformat()
 
 
-def uuid_generator():
+def uuid_generator() -> str:
     """
-    Returns a unique UUID
-    ex. '0000a0a0-a000-00a0-00a0-0000a0000000'
+    Generates a unique UUID.
+
+    Returns:
+        A string representing a UUID.
+        ex. '0000a0a0-a000-00a0-00a0-0000a0000000'
     """
     my_uuid = uuid.uuid4()
     return str(my_uuid)
 
-def result_to_json(result_rows):
+
+def result_to_json(result_rows):  # todo: return type
     """
     Converts a database result to Json format
+
+    Args:
+        result_rows: The object to convert to json.
+
+    Returns: todo
     """
-    json_result = json.loads(json.dumps(result_rows, default=myconverter))
+    json_result = json.loads(json.dumps(result_rows, default=myconverter))  # todo: wha?
     return json_result
 
 
-def myconverter(obj):       #pylint: disable-msg=inconsistent-return-statements
+# todo: Figure out how the README.md is generated.
```

Comment:
Which `README.md` are you talking about here?

Reply:
The one in pg_utils. Contains documentation on each function, so I'm not sure if I'd have to manually update it or if it's generated in some way,